### ISSUE TITLE
Print firmware revision in Settings

### ISSUE THREAD

**Issue by tsusanka** [trezor/trezor-suite#2917]
In Suite we print its version and we make it clickable. It goes to Github on the particular commit.

![image](https://user-images.githubusercontent.com/1835345/98520952-155e0a00-2273-11eb-8a9c-10a726ba03c9.png)

We could do the very same for firmware. We currently print only the version but we could also make the version clickable and link it to the particular firmware's commit.

![image](https://user-images.githubusercontent.com/1835345/98520921-0bd4a200-2273-11eb-96c4-d4241ddd2ea9.png)

----

Firmware prints this information in Features.

```
$ trezorctl get-features | grep revision                    
    revision: 9 bytes b'fd90bb9d1',
```

This might simplify debugging and bug reporting.

**Comment by waltonzt** [comment]
@tsusanka I would like to take this issue on. I have reviewed the work required and the only thing I am missing is how to redirect to a commit in the trezor-firmware repo. The revision number in features does not seem to be related to the firmware commits. 

The Suite version code looks to use an env variable COMMITHASH to get the needed values. 

Would you have any guidance on where to git the commit hash for trezor-firmware?

**Comment by prusnak** [comment]
> The revision number in features does not seem to be related to the firmware commits.

It is. From the above example: `fd90bb9d1` should link to `https://github.com/trezor/trezor-firmware/commit/fd90bb9d1`

**Comment by waltonzt** [comment]
Okay that is where I am confused then because when I access `device.features.revision` I do not get a 9 character commit hash. Accessing that value returns a 18 character string. when appending that to the end of the url like you have in your example it does not go to a valid commit. 

Is this due to it being a dev environment?

**Comment by prusnak** [comment]
> Accessing that value returns a 18 character string.

I think I know what's the issue - Javascript does not have native bytes type, so we hex-encode protobuf bytes into a Javascript string. Thus you you need to hex-decode the 18 character string into a 9-byte value first. Not sure what's the best way from top off my head, but surely there are other instances of such conversion in the existing code.

We use bytes instead of string for revision field for historical reasons.

**Comment by waltonzt** [comment]
That was it! Thank you for your help

**Comment by tsusanka** [comment]
Closed by #3629. Thanks a lot for your contribution @waltonzt!
